lib/jekyll/commands/console.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "irb"
+
+module Jekyll
+  module Commands
+    class Console < Command
+      class << self
+        def init_with_program(prog)
+          prog.command(:console) do |c|
+            c.syntax "console"
+            c.description "Invoke an IRB console with the site loaded"
+            c.alias :c
+
+            c.option "config", "--config CONFIG_FILE[,CONFIG_FILE2,...]", Array,
+                     "Custom configuration file"
+
+            c.action do |_, options|
+              Jekyll::Commands::Console.process(options)
+            end
+          end
+        end
+
+        # TODO: is there a way to add a unit test for this command?
+        # rubocop:disable Style/GlobalVars, Metrics/AbcSize, Metrics/MethodLength
+        def process(options)
+          Jekyll.logger.info "Starting:", "Jekyll v#{Jekyll::VERSION.magenta}" \
+                                      " console…"
+          Jekyll.logger.info "Environment:", Jekyll.env.cyan
+          site = Jekyll::Site.new(configuration_from_options(options))
+          site.reset
+          site.read
+          site.generate
+
+          $JEKYLL_SITE = site
```

Comment:
Is this global variable absolutely necessary?
Could `site` be passed into the `workspace` instead?

Reply:
Great question—I tried a bunch of things to avoid having to use a global variable and wasn't successful. The challenge is it's setting up the REPL to run within the top-level `main` context, which I think is what we want, but once there I don't have access to any previously-defined local variables like `site`. Any suggestions?

Reply:
To be honest, I don't have much experience with setting up REPL environments.
Will have to dig into the Ruby docs..